AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/managespace/ManageSpaceFragment.kt

```diff
@@ -0,0 +1,386 @@
+        flowOfDeleteUnusedMediaSize.ifCollectionDirectoryExistsEmit {
+            withCol {
+                val unusedFiles = media.findUnusedMediaFiles()
+                val unusedFilesSize = unusedFiles.sumOf { file -> file.calculateSize() }
+                Size.FilesAndBytes(unusedFiles, unusedFilesSize)
+            }
+        }
+    }
+
+    suspend fun performMediaCheck() {
+        withCol { media.performFullCheck() }
+
+        launchSearchForUnusedMedia()
+    }
+
+    suspend fun deleteMedia(filesNamesToDelete: List<String>) {
+        withCol { deleteMedia(this, filesNamesToDelete) }
+
+        launchCalculationOfSizeOfEverything()
+        launchCalculationOfCollectionSize()
+        launchSearchForUnusedMedia()
+    }
+
+    /***************************************** Backups ********************************************/
+
+    private fun launchCalculationOfBackupsSize() = viewModelScope.launch {
+        flowOfDeleteBackupsSize.ifCollectionDirectoryExistsEmit {
+            withCol {
+                val backupFiles = BackupManager.getBackups(File(this.path)).toList()
+                val backupFilesSize = backupFiles.sumOf { file -> file.calculateSize() }
+                Size.FilesAndBytes(backupFiles, backupFilesSize)
+            }
+        }
+    }
+
+    suspend fun deleteBackups(backupsToDelete: List<File>) {
+        withCol { BackupManager.deleteBackups(this, backupsToDelete) }
+
+        launchCalculationOfBackupsSize()
+        launchCalculationOfCollectionSize()
+        launchCalculationOfSizeOfEverything()
+    }
+
+    /*************************************** Collection *******************************************/
+
+    private fun launchCalculationOfCollectionSize() = viewModelScope.launch {
+        flowOfDeleteCollectionSize.ifCollectionDirectoryExistsEmit {
+            withContext(Dispatchers.IO) {
+                Size.Bytes(collectionDirectory.calculateSize())
+            }
+        }
+    }
+
+    suspend fun deleteCollection() {
+        withQueue { CollectionManager.deleteCollectionDirectory() }
+
+        launchCalculationOfBackupsSize()
+        launchCalculationOfSizeOfEverything()
+        launchCalculationOfCollectionSize()
+        launchSearchForUnusedMedia()
+    }
+
+    /*************************************** Everything *******************************************/
+
+    // Retrieving app data size before Oreo is too complicated, skip for simplicity.
+    // See https://stackoverflow.com/a/36983630
+    private fun launchCalculationOfSizeOfEverything() = viewModelScope.launch {
+        flowOfDeleteEverythingSize.emit(Size.Calculating)
+        flowOfDeleteEverythingSize.emit(
+            withContext(Dispatchers.IO) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    Size.Bytes(AnkiDroidApp.instance.getUserDataAndCacheSize())
+                } else {
+                    Size.Error(Exception("Not implemented"), widgetText = "🤷️")
```

Comment:
:thinking: 

```suggestion
                    Size.Error(Exception("Not implemented"), widgetText = "Size is only calculated for Android 8 and higher")
```

...and, needs extraction? Not sure where this will show up or if it will

Reply:
Widget is that little box on the right-hand side of the preference, the text should be reeeal short, hence emoji. Not sure what to put here. Or maybe we should bother to calculate, it *is* possible

Reply:
:thinking:  Could simply go with `?` for now + defer task to follow-on, it won't affect strings nor will it affect structure of everything so seems like deferrable work

Reply:
Maybe “N/A” for now? And let me see how exactly problematic the calculation will be

Reply:
N/A works for me